bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetBridgeHandler.java

```diff
@@ -0,0 +1,204 @@
+        return Collections.singleton(JuiceNetDiscoveryService.class);
+    }
+
+    public void handleApiException(Exception e) {
+        if (e instanceof JuiceNetApiException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.toString());
+        } else if (e instanceof InterruptedException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.toString());
+            Thread.currentThread().interrupt();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.NONE, e.toString());
+        }
+    }
+
+    @Nullable
+    public Thing getThingById(String id) {
+        List<Thing> childThings = getThing().getThings();
+
+        for (Thing childThing : childThings) {
+            Configuration configuration = childThing.getConfiguration();
+
+            String childId = configuration.get(PARAMETER_UNIT_ID).toString();
+
+            if (childId.equals(id)) {
+                return childThing;
+            }
+        }
+
+        return null;
+    }
+
+    // This function will query the list of devices from the API and then set the name/token in the child handlers. If a
+    // child does not exist, it will notify the Discovery service. If it is successful, it will ensure the bridge status
+    // is updated
+    // to ONLINE.
+    public void iterateApiDevices() {
+        List<JuiceNetApiDevice> listDevices;
+        JuiceNetDiscoveryService discoveryService = this.discoveryService;
+
+        if (discoveryService == null) {
+            return;
+        }
+
+        try {
+            listDevices = api.queryDeviceList();
+        } catch (JuiceNetApiException | InterruptedException e) {
+            handleApiException(e);
+            return;
+        }
+
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        for (JuiceNetApiDevice dev : listDevices) {
+            Thing childThing = getThingById(dev.unitId);
+            if (childThing == null) {
+                discoveryService.notifyDiscoveryDevice(dev.unitId, dev.name);
+            } else {
+                JuiceNetDeviceHandler childHandler = (JuiceNetDeviceHandler) childThing.getHandler();
+                if (childHandler != null) {
+                    childHandler.setNameAndToken(dev.name, dev.token);
+                }
+            }
+        }
+    }
+
+    private void pollDevices() {
+        logger.debug("pollDevices");
+
+        List<Thing> things = getThing().getThings();
+
+        for (Thing t : things) {
+            if (!t.getThingTypeUID().equals(DEVICE_THING_TYPE)) {
+                continue;
+            }
+
+            JuiceNetDeviceHandler handler = (JuiceNetDeviceHandler) t.getHandler();
+            if (handler == null) {
+                logger.trace("no handler");
+                return;
```

Comment:
```suggestion
                continue;
```

or is it intentional to skip other things?